src/diffpy/utils/diffraction_objects.py

```diff
@@ -263,12 +263,30 @@ def get_array_index(self, xtype, value):
         -------
         the index of the value in the array
         """
-        if self.on_xtype(xtype) is None:
-            raise ValueError(_xtype_wmsg(xtype))
-        for i, target in enumerate(self.on_xtype(xtype)[0]):
-            if value == target:
-                return i
-        raise IndexError(f"WARNING: no matching value {value} found in the {xtype} array.")
+
+        if xtype is None:
+            xtype = self.input_xtype
+        if self.on_xtype(xtype) is None or len(self.on_xtype(xtype)[0]) == 0:
+            raise ValueError(
+                f"The '{xtype}' array is empty. " "Please ensure it is initialized and the correct xtype is used."
```

Comment:
The current `on_xtype` function does not raise an error when the specified xtype is invalid. I was thinking about letting that function raise an error in those cases, but it looks like we can handle invalid xtype together with empty array.